Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteGPUProxy.cpp

```diff
@@ -43,64 +43,41 @@
 
 namespace WebKit {
 
-RefPtr<RemoteGPUProxy> RemoteGPUProxy::create(GPUProcessConnection& gpuProcessConnection, WebGPU::ConvertToBackingContext& convertToBackingContext, WebGPUIdentifier identifier, RenderingBackendIdentifier renderingBackend)
+RefPtr<RemoteGPUProxy> RemoteGPUProxy::create(RemoteRenderingBackendProxy& remoteRenderingBackendProxy, WebGPU::ConvertToBackingContext& convertToBackingContext, WebGPUIdentifier identifier, RenderingBackendIdentifier renderingBackend)
 {
     constexpr size_t connectionBufferSizeLog2 = 21;
     auto [clientConnection, serverConnectionHandle] = IPC::StreamClientConnection::create(connectionBufferSizeLog2);
     if (!clientConnection)
         return nullptr;
-    auto remoteGPUProxy = adoptRef(new RemoteGPUProxy(gpuProcessConnection, clientConnection.releaseNonNull(), convertToBackingContext, identifier));
+    auto remoteGPUProxy = adoptRef(new RemoteGPUProxy(remoteRenderingBackendProxy, clientConnection.releaseNonNull(), convertToBackingContext, identifier));
     remoteGPUProxy->initializeIPC(WTFMove(serverConnectionHandle), renderingBackend);
     return remoteGPUProxy;
 }
 
 
-RemoteGPUProxy::RemoteGPUProxy(GPUProcessConnection& gpuProcessConnection, Ref<IPC::StreamClientConnection> clientConnection, WebGPU::ConvertToBackingContext& convertToBackingContext, WebGPUIdentifier identifier)
+RemoteGPUProxy::RemoteGPUProxy(RemoteRenderingBackendProxy& remoteRenderingBackendProxy, Ref<IPC::StreamClientConnection> clientConnection, WebGPU::ConvertToBackingContext& convertToBackingContext, WebGPUIdentifier identifier)
     : m_backing(identifier)
     , m_convertToBackingContext(convertToBackingContext)
-    , m_gpuProcessConnection(&gpuProcessConnection)
+    , m_remoteRenderingBackendProxy(remoteRenderingBackendProxy)
     , m_streamConnection(WTFMove(clientConnection))
 {
 }
 
 RemoteGPUProxy::~RemoteGPUProxy()
 {
-    disconnectGpuProcessIfNeeded();
+    m_streamConnection->invalidate();
+    m_remoteRenderingBackendProxy.streamConnection().send(Messages::RemoteRenderingBackend::ReleaseRemoteGPU(m_backing), m_remoteRenderingBackendProxy.renderingBackendIdentifier(), Seconds::infinity());
 }
 
 void RemoteGPUProxy::initializeIPC(IPC::StreamServerConnection::Handle&& serverConnectionHandle, RenderingBackendIdentifier renderingBackend)
 {
-    m_gpuProcessConnection->addClient(*this);
```

Comment:
These changes seem fine but I will defer to @kkinnunen-apple as I see RemoteMedia/Audio/Image/etc using the `addClient` pattern unlike WebGL